homeassistant/components/xiaomi_miio/select.py

```diff
@@ -0,0 +1,187 @@
+
+    _LOGGER.debug("Initializing with host %s (token %s...)", host, token[:5])
+
+    if model in [MODEL_AIRHUMIDIFIER_CA1, MODEL_AIRHUMIDIFIER_CB1]:
+        entity_class = XiaomiAirHumidifierSelector
+    elif model in [MODEL_AIRHUMIDIFIER_CA4]:
+        entity_class = XiaomiAirHumidifierMiotSelector
+    elif model in MODELS_HUMIDIFIER:
+        entity_class = XiaomiAirHumidifierSelector
+    else:
+        return
+
+    for feature in SELECTOR_TYPES:
+        selector = SELECTOR_TYPES[feature]
+        entities.append(
+            entity_class(
+                f"{config_entry.title} {selector.name}",
+                device,
+                config_entry,
+                f"{selector.short_name}_{config_entry.unique_id}",
+                selector,
+                coordinator,
+            )
+        )
+
+    async_add_entities(entities)
+
+
+class XiaomiSelector(XiaomiCoordinatedMiioEntity, SelectEntity):
+    """Representation of a generic Xiaomi attribute selector."""
+
+    def __init__(self, name, device, entry, unique_id, selector, coordinator):
+        """Initialize the generic Xiaomi attribute selector."""
+        super().__init__(name, device, entry, unique_id, coordinator)
+        self._attr_icon = selector.icon
+        self._controller = selector
+        self._attr_options = self._controller.options
+
+    @staticmethod
+    def _extract_value_from_attribute(state, attribute):
+        value = getattr(state, attribute)
+        if isinstance(value, Enum):
+            return value.value
+
+        return value
+
+
+class XiaomiAirHumidifierSelector(XiaomiSelector):
+    """Representation of a Xiaomi Air Humidifier selector."""
+
+    def __init__(self, name, device, entry, unique_id, controller, coordinator):
+        """Initialize the plug switch."""
+        super().__init__(name, device, entry, unique_id, controller, coordinator)
+        self._current_led_brightness = self._extract_value_from_attribute(
+            self.coordinator.data, self._controller.short_name
+        )
+
+    @callback
+    def _handle_coordinator_update(self):
+        """Fetch state from the device."""
+        self._current_led_brightness = self._extract_value_from_attribute(
+            self.coordinator.data, self._controller.short_name
+        )
+        self.async_write_ha_state()
+
+    @property
+    def current_option(self):
+        """Return the current option."""
+        return self.led_brightness
+
+    async def async_select_option(self, option: str) -> None:
+        """Set an option of the miio device."""
+        if option not in self.options:
+            raise ValueError(
+                f"Selection '{option}' is not a valid {self._controller.name}"
+            )
+        await self.async_set_led_brightness(option)
+
+    @property
+    def led_brightness(self):
+        """Return the current led brightness."""
+        reversed_value_map = {0: "Bright", 1: "Dim", 2: "Off"}
```

Comment:
This dict can be extracted to a constant.

Reply:
Added a const